packages/alfa-rules/src/sia-r83/rule.ts

```diff
@@ -134,29 +137,55 @@ function isPossiblyClipped(device: Device): Predicate<Element> {
   };
 }
 
-function wrapsText(device: Device): Predicate<Text> {
-  return (text) =>
-    text
-      .ancestors({
-        flattened: true,
-      })
-      .filter(isElement)
-      .find(isPossiblyClipped(device))
-      .some((element) => {
-        const style = Style.from(element, device);
-
-        const { value: whitespace } = style.computed("white-space");
-
-        // If whitespace does not cause wrapping, we need to check if a text
-        // overflow could cause the text to clip.
-        if (whitespace.value === "nowrap") {
-          const { value: overflow } = style.computed("text-overflow");
-
-          // We assume that the text won't clip if the text overflow is handled
-          // any other way than clip.
-          return overflow.value !== "clip";
-        }
+function wrapsText(device: Device): Predicate<Element> {
+  return (element) => {
+    if (isPossiblyClipped(device)(element)) {
+      return isActuallyClipping(element, device);
+    }
+
+    const relevantParent = isPositioned(device, "static")(element)
+      ? element.parent().filter(isElement)
+      : offsetParent(element, device);
+
+    return relevantParent.every(wrapsText(device));
+  };
+}
+
+function isPositioned(device: Device, position: string): Predicate<Element> {
+  return (element) =>
+    Style.from(element, device).computed("position").value.value === position;
+}
+
+const isActuallyClipping = (element: Element, device: Device) => {
```

Comment:
I think this is actually checking that the element is actually **not** clipping and should be renamed `isNotActuallyClipping`. (if there is an `overflow` of `clip`, then it returns `false`…)
(or we keep the name and negate the returns, and use a wrapping `not()` at call site)

Reply:
Also, `isActuallyClipping` should be a `function` declaration.

Reply:
These haven't been addressed.
Should be 
```javascript
function isNotActuallyClipping(element: Element, device: Device): boolean {
```
Or maybe, since it's only used once and negated:
```javascript
function isActuallyClipping(element: Element, device: Device): boolean {
```
plus negating the `return` + removing the `!` when calling it.